cypher-shell/src/main/java/org/neo4j/shell/cli/InteractiveShellRunner.java

```diff
@@ -145,10 +157,24 @@ AnsiFormattedText getPrompt() {
         if (statementParser.containsText()) {
             return continuationPrompt;
         }
-        if (txHandler.isTransactionOpen()) {
-            return transactionPrompt;
-        }
-        return freshPrompt;
+
+        String databaseName = databaseManager.getActiveDatabase();
+
+        // Substitute empty name for the default default-database-name
+        // For now we just use a hard-coded default name
+        // Ideally we would like to receive the actual name in the ResultSummary when we connect (in BoltStateHandler.reconnect())
+        // (If the user is an admin we could also query for the default database config value with:
```

Comment:
I thought you said we were potentially aiming at replacing all built-in dbms-procedures that access the system database with DDL commands?
The existing implementation of `dbms.listConfig` does not need to access system the database (but you still have to be admin). This will soon change when we start moving config into system database, but then we may could also create a DDL command for that, e.g. `SHOW CONFIG`.

Reply:
I checked with PM, and we can run `SHOW DATABASES` as a normal read-user and do not need administrator privileges. All other system commands are update commands, and update on systemdb needs administrator privilages.